stellargraph/core/element_data.py

```diff
@@ -344,10 +347,12 @@ def _init_directed_adj_lists(self):
             out_dict.setdefault(src, []).append(i)
 
         dtype = np.min_scalar_type(len(self.sources))
-        self._edges_in_dict = _numpyise(in_dict, dtype=dtype)
-        self._edges_out_dict = _numpyise(out_dict, dtype=dtype)
+        return _numpyise(in_dict, dtype=dtype), _numpyise(out_dict, dtype=dtype)
 
     def _init_undirected_adj_lists(self):
+        self._edges_dict = self._create_undirected_adj_lists()
+
+    def _create_undirected_adj_lists(self):
```

Comment:
why do we need this new function?

Reply:
For the memory benchmarks. In particular, measuring the memory usage requires creating the to-be-measured objects within the benchmarked function, and returning them. This is fine for something like creating a `StellarGraph`, because we can just do that repeatedly, however it doesn't work quite so well with these sort of lazy properties.

Calling `_init_undirected_adj_lists` several times on a single `StellarGraph` won't work, because it'll deallocate the previous adjacency lists, resulting in a total difference of 0 (or close to), for all but the first run. (That is, it'll start with `self._edges_dict = adj_list_1`, then effectively run `del self._edges_dict; self._edges_dict = adj_list_2`, where `adj_list_1` and `adj_list_2` have the same content, but are different objects.)

By having this as a "pure" function that just computes the adjacency lists and returns them, we can pass the result around to measure its memory usage.

Another option would be to create a new `StellarGraph` object for each benchmark run, outside the timing function, but this is slower and quite fiddly.

Does that vaguely make sense?